aead/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+//! A set of traits designed to support authenticated encryption.
+
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
If we are fine making 1.36 the default, we can eliminate the std feature completely and just use `alloc`.

Aside: if it ever needs to come back for some reason, my own recommendation is that the feature be `no_std`, which would be enabled by default. It's semantically weird, but it has the advantage of making things Just Work when some dep deep in your tree inevitably forgets to do default-features = false.

Reply:
This violates Rust API guidelines, which are to use an additive feature named `std`:

https://rust-lang-nursery.github.io/api-guidelines/naming.html#feature-names-are-free-of-placeholder-words-c-feature

If you want to avoid `default-features = false`, don't make `std` a default feature.

Reply:
Well-OT, but I think those guidelines are misguided, in this specific case. The union of features + one crate that leaves it on by default *cough*serde* + one inattentive intermediate dependency = weird build failure in an unrelated crate (in my case, usually a custom panic handler).

If it was `no_std`, at least the inattentive intermediate dependency would cause a build fail in the crate that's trying to use the "std" features of the now no-std crate, and it doesn't need `cargo-nono` to isolate.

Reply:
As someone who's been a heavy `no_std` Rust user for several years, I empathize with these concerns.

However, I also feel like they're completely addressed by leaving `std` out of the default features.

Cargo features are supposed to be additive. Having functionality go away by enabling a Cargo feature very much goes against the grain of what "features" are supposed to be. Among other things, it means you lose the ability to express which other features might be dependent on the `std` feature.

Reply:
I'll note that negative-polarity features (i.e. enabling them reduces API surface) and mixed-polarity features (enabled and disabled expose incompatible APIs) actively break Cargo's ability to resolve dependencies, and cause spooky build failures; see more complete explanation in https://github.com/m4b/goblin/issues/8#issuecomment-253703418